islandora_pathauto.module

```diff
@@ -251,3 +253,87 @@ function islandora_pathauto_tokens($type, $tokens, array $data = array(), array
   }
   return array();
 }
+
+/*
+ * Implements hook_form_alter()
+ * 
+ * Adds a TextField to Islandora Properties so the User can Change
+ * the URL Alias.
+ */
+function islandora_pathauto_form_alter(&$form , &$form_state, $form_id){
+    if( $form_id == 'islandora_object_properties_form' ){
+        $generate_url = variable_get( 'set_generate', 0 );
+        
+        $form['check'] = array(
+            '#type'          => 'checkbox',
+            '#title'         => t('Generate automatic URL alias'),
+            '#default_value' => $generate_url,
+            '#description'   => t('Uncheck this to create a custom alias below.'),
+            '#weight'        => -1,
+        );
+        
+        $original_path = 'islandora/object/' . $form['pid']['#value'];
+        if( drupal_get_path_alias($original_path) === $original_path ){
+            $url_alias = '';
+        }
+        else{
+            $url_alias = drupal_get_path_alias($original_path);  
+        }
+        
+        $form['object_url'] = array(
+            '#title'         => t('URL Alias'),
+            '#default_value' => $url_alias,
+            '#required'      => FALSE,
+            '#description'   => t("An alternative URL Alias"),
+            '#type'          => 'textfield',
+            '#weight'        => -1,
+            '#states' => array(
+			"disabled" => array(
+                            "input[name='check']" => array("checked" => TRUE)),
+			),
+        );
+        array_push( $form['#submit'], 'islandora_pathauto_form_submit');
+    }
+}
+
+/*
+ * Implements form_submit() for the Custom URL Alias Option
+ * 
+ * Submits the user's alias to change the islandora Object URL path
+ */
+function islandora_pathauto_form_submit($form, &$form_state) { 
+   $original_path = 'islandora/object/' . $form['pid']['#value'];
+    $new_path = path_load($original_path);
+    $new_path['alias'] = $form_state['values']['object_url'];
+    $new_path['source'] = $original_path;
+    
+    if( $form_state['values']['check'] === 1){
+        $new_path['alias'] = 'myislandoraobjects/' . $form_state['values']['object_label'];
```

Comment:
myislandoraobjects/ is for the generate url aliases. I hardcorded it because I didn't know what path I should use for it. I could change it now.